Common/src/main/java/com/almostreliable/unified/utils/JsonCompare.java

```diff
@@ -270,6 +268,17 @@ public String getName() {
         }
     }
 
+    public record CompareContext(CompareSettings settings, List<String> compareFields) {
```

Comment:
Is there a specific reason you are using a list instead of a set here? Could even be a collection or iterable only. I am just asking because inside the `create` method, you retrieve the recipe key set and when the ignored fields are not empty, you copy the collection twice. Once inside the `new HashSet` and then with `List.copyOf` inside the CTor call. Is is because you want to make it immutable again?

Reply:
The PR is focused on efficiency of the hot path, which in this case (according to profiling) is the path that iterates over the `compareFields`. We can sloppily make lots of copies elsewhere and it'll take less than 1ms in total, but if we iterate over a `HashSet` in a hot loop it may take 100ms longer than a `List` for example, so the main requirement is to enforce `compareFields` is a type that is fast to iterate over.

The `Set` is used earlier because it's a quick way to remove any duplicates and remove the ignored fields. You can do the same thing in many ways but there are tradeoffs between copying, `contains` performance, and clarity. Since the code there isn't called as much, I think I just tried to make it easy to read.